zone/worldserver.cpp

```diff
@@ -2107,66 +2352,447 @@ void WorldServer::HandleMessage(uint16 opcode, const EQ::Net::Packet &p)
+		auto client_group = entity_list.GetGroupByID(CZUA->group_id);
+		if (client_group) {
+			for (int member_index = 0; member_index < MAX_GROUP_MEMBERS; member_index++) {
+				if (client_group->members[member_index] && client_group->members[member_index]->IsClient()) {
+					auto group_member = client_group->members[member_index]->CastToClient();
+					group_member->FailTask(CZUA->task_id);
+				}
+			}
+		}
+		break;
+	}
+	case ServerOP_CZTaskFailRaid:
+	{
+		CZTaskFailRaid_Struct* CZUA = (CZTaskFailRaid_Struct*) pack->pBuffer;
+		auto client_raid = entity_list.GetRaidByID(CZUA->raid_id);
+		if (client_raid) {
+			for (int member_index = 0; member_index < MAX_RAID_MEMBERS; member_index++) {
+				if (client_raid->members[member_index].member && client_raid->members[member_index].member->IsClient()) {
+					auto raid_member = client_raid->members[member_index].member->CastToClient();
+					raid_member->FailTask(CZUA->task_id);
+				}
+			}
+		}
+		break;
+	}
+	case ServerOP_CZTaskFailGuild:
+	{
+		CZTaskFailGuild_Struct* CZUA = (CZTaskFailGuild_Struct*) pack->pBuffer;
+		auto client_list = entity_list.GetClientList();
+		for (auto client : client_list) {
+			if (client.second->GuildID() > 0 && client.second->GuildID() == CZUA->guild_id) {
+				client.second->FailTask(CZUA->task_id);
+			}
+		}
+		break;
+	}
+	case ServerOP_CZTaskRemovePlayer:
+	{
+		CZTaskRemovePlayer_Struct* CZTR = (CZTaskRemovePlayer_Struct*) pack->pBuffer;
+		auto client = entity_list.GetClientByCharID(CZTR->character_id);
+		if (client != 0) {
+			client->RemoveTaskByTaskID(CZTR->task_id);
+		}
+		break;
+	}
+	case ServerOP_CZTaskRemoveGroup:
+	{
+		CZTaskRemoveGroup_Struct* CZTR = (CZTaskRemoveGroup_Struct*) pack->pBuffer;
+		auto client_group = entity_list.GetGroupByID(CZTR->group_id);
+		if (client_group) {
+			for (int member_index = 0; member_index < MAX_GROUP_MEMBERS; member_index++) {
+				if (client_group->members[member_index] && client_group->members[member_index]->IsClient()) {
+					auto group_member = client_group->members[member_index]->CastToClient();
+					group_member->RemoveTaskByTaskID(CZTR->task_id);
+				}
+			}
+		}
+		break;
+	}
+	case ServerOP_CZTaskRemoveRaid:
+	{
+		CZTaskRemoveRaid_Struct* CZTR = (CZTaskRemoveRaid_Struct*) pack->pBuffer;
+		auto client_raid = entity_list.GetRaidByID(CZTR->raid_id);
+		if (client_raid) {
+			for (int member_index = 0; member_index < MAX_RAID_MEMBERS; member_index++) {
+				if (client_raid->members[member_index].member && client_raid->members[member_index].member->IsClient()) {
+					auto raid_member = client_raid->members[member_index].member->CastToClient();
+					raid_member->RemoveTaskByTaskID(CZTR->task_id);
+				}
+			}
+		}
+		break;
+	}
+	case ServerOP_CZTaskRemoveGuild:
+	{
+		CZTaskRemoveGuild_Struct* CZTR = (CZTaskRemoveGuild_Struct*) pack->pBuffer;
+		auto client_list = entity_list.GetClientList();
+		for (auto client : client_list) {
```

Comment:
Not suggesting a change here; but you can inline `for (auto client : entity_list.GetClientList()) {`